### PR TITLE
[FLINK-2512]Add client.close() before throw RuntimeException

### PR DESCRIPTION
**Pull request by ffbin**
In line 129, it close client in finally{} before throw exception.
But in line 105, it throw exception without close client.
So I think it is better to close client before throw RuntimeException.

### CODE REVIEW

**Comment by uce** [comment]
I think the best thing would be to just move the `try` up a little.

**Comment by ffbin** [comment]
@uce Thanks. What about remove "if(client.getTopologyJobId(name) != null) {...}" in line 103, because submitTopologyWithOpts() has check it at the head of function and will throw AlreadyAliveException.Then in finally{} close client.

**Comment by uce** [comment]
If you move the try up, you can certainly remove the manual close. Regarding the check in 103, it really depends on whether the Strom compat layer depends on having only a single job per client. Therefore I would keep it in and let it throw the RuntimeException as before. The finally block will then make sure that the client is closed.

**Comment by hsaputra** [comment]
Since ``getTopologyJobId`` can also throw exception, we could just move up the try-catch to include the call to that method and rely on finally to close the client.

**Comment by ffbin** [comment]
@uce @hsaputra  Thanks. I have move the try up and rely on finally to close the client.

**Comment by StephanEwen** [comment]
Looks good, will merge this...

**Comment by hsaputra** [comment]
All tests are failing though, not sure if bc this patch

File: flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkSubmitter.java [hunk 1]
```diff
@@ -99,23 +99,24 @@ public static void submitTopology(final String name, final Map stormConf, final
...
-		}
-		String localJar = System.getProperty("storm.jar");
-		if (localJar == null) {
-			try {
-				for (final File file : ((ContextEnvironment) ExecutionEnvironment.getExecutionEnvironment())
-						.getJars()) {
-					// TODO verify that there is onnly one jar
-					localJar = file.getAbsolutePath();
+		try {
+			final FlinkClient client = FlinkClient.getConfiguredClient(stormConf);
```
Comment (hsaputra) [inline flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkSubmitter.java]: Shouldn't declaration of ``client`` be outside of the try clause?

**Comment by StephanEwen** [comment]
You are right, CI caught it. From my skimming over, the code looked good, but it does not compile...

**Comment by ffbin** [comment]
@hsaputra @StephanEwen  Thanks. I have fix my code error.

**Comment by hsaputra** [comment]
+1

**Pull request closed without merging**